### DISCUSSION TITLE
Añadir un STEP para generar numeros aleatorios

### DISCUSSION THREAD

**Discussion by rommelayala** [veepee-oss/gingerspec#55] [Ideas]
Agradeceria que añadan un metodo de generacion de numeros aleatorios y no tener que crearlo como un custom STEP, Jose ya me lo ha creado solo falta incluirlo en la libreria.

**Comment by josefd8** [comment]
Hey @rommelayala 

Such step could be included in the next release (2.1.6). Could you elaborate a bit more on what should be the syntax of the step? for example:

- Given I create a random number between '0' and '255' and save it in the variable 'RANDOM'

- GIven I create a random alphanumeric string of '8' chars and save it 'RANDOM'

- GIven I create a random number of '8' chars and save it 'RANDOM'

......etc

Could you provide some feedback about what other use cases could be useful for you ?

**Comment by rommelayala** [comment]
La primera opcion me parace mas adecuada en este momento.
The first one will be useful at this moment.

**Comment by josefd8** [comment]
Hey @rommelayala 

In version 2.2.0-RC3, this new functionality was added:

- Generate a random number within range:
  Example: 

> Given I generate a random number between '0' and '10' and save it in the variable 'RANDOM'

- Generate a random alphanumeric string of given length
  Example: 

> Given I generate a random 'alphanumeric' string of length '20' and save it in the variable 'ALPHANUMERIC'

- Generate a random numeric string of given length
  Example: 

> Given I generate a random 'numeric' string of length '20' and save it in the variable 'NUMERIC'


Alternatively, remember that you can also execute commands in the host shell, so if you know a command to generate random numbers you could do something like

> Given I run 'jot -r 1  2000 65000' locally and save the value in environment variable 'RANDOM'

(I tested this on OSX but there must be an equivalent for Linux).

Also, starting from GingerSpec v2.2.0-RC3, you can execute javascript functions in variables using [StringSubstitutor](https://commons.apache.org/proper/commons-text/apidocs/org/apache/commons/text/StringSubstitutor.html), for example:

> ${script:javascript:Math.floor(Math.random() * 10)}

(This is for generating a value from 0 to 9. This is a simple javascript function I found on the Internet, is just an example)

For more info on how to use the new type of GingerSpec variables please read [this](https://github.com/vpTechOSS/gingerspec/discussions/61) discussion
